pkg/driver/docker/docker.go

```diff
@@ -284,10 +284,31 @@ func (c dockerClient) StartBuildkitd(ctx context.Context, tag, name string, bc *
 	}
 	created, _ := c.Exists(ctx, name)
 	if created {
-		err := c.ContainerStart(ctx, name, types.ContainerStartOptions{})
+		status, err := c.GetStatus(ctx, name)
 		if err != nil {
-			return name, err
+			return name, errors.Wrap(err, "failed to get container status")
 		}
+
+		if status == "paused" {
+			logger.Info("container was paused, unpause it now...")
+			_, err := c.ResumeContainer(ctx, name)
+			if err != nil {
+				return name, errors.Wrap(err, "failed to unpause container")
+			}
+		} else if status == "exited" {
+			logger.Info("container exited, try to restart it...")
+			err := c.ContainerRestart(ctx, name, container.StopOptions{})
+			if err != nil {
+				return name, errors.Wrap(err, "failed to restart cotaniner")
+			}
+		} else {
+			logger.Info("container already exists.")
```

Comment:
There are still several other possible statuses here: https://docs.docker.com/engine/reference/commandline/ps/#status

Reply:
Do we need to deal with all of them?

> "created", "running", "paused", "restarting", "removing", "exited", or "dead"

If the status is `created` or `running`, its control flow goes into `else` branch, just starting container.

If the status is `paused`, I unpause the container.

If the status is `exited`, I restart the container.

The remaining statues are `removing` and `dead`, it's rare and I think maybe we does not need to consider of it?

Reply:
Could we make these statuses like this?

```
  type ContainerStatus string
```

Reply:
Sure!

Reply:
Please add some comments.